packages/common/src/store/pages/chat/middleware.ts

```diff
@@ -0,0 +1,73 @@
+import { ChatEvents, type sdk } from '@audius/sdk'
+import { Middleware } from 'redux'
+
+import { actions as chatActions } from './slice'
+
+const {
+  connect,
+  disconnect,
+  addMessage,
+  incrementUnreadCount,
+  setMessageReactionSucceeded
+} = chatActions
+
+export const chatMiddleware =
+  (audiusSdk: () => Promise<ReturnType<typeof sdk>>): Middleware =>
+  (store) => {
+    let messageListener: ChatEvents['message'] | null = null
+    let reactionListener: ChatEvents['reaction'] | null = null
+    let openListener: ChatEvents['open'] | null = null
+    let closeListener: ChatEvents['close'] | null = null
+
+    return (next) => (action) => {
+      if (connect.match(action)) {
+        console.log('Chat middleware attaching')
+        audiusSdk().then((sdk) => {
+          openListener = () => {
+            console.log(
+              'Chat middleware WebSocket opened. Listening for chats...'
+            )
+          }
+          messageListener = ({ chatId, message }) => {
+            store.dispatch(addMessage({ chatId, message }))
+            store.dispatch(incrementUnreadCount({ chatId }))
+          }
+          reactionListener = ({ chatId, messageId, reaction }) => {
+            store.dispatch(
+              setMessageReactionSucceeded({
+                chatId,
+                messageId,
+                reaction
+              })
+            )
+          }
+          closeListener = () => {
+            console.log('Chat middleware WebSocket closed. Reconnecting...')
+            sdk.chats.listen()
```

Comment:
Might need some more clever logic here to not spam forever when a DN goes down or becomes unreachable, and eventually reselect...